pkg/analysis/passes/osvscanner/osvscanner.go

```diff
@@ -0,0 +1,222 @@
+	scanningFailure                    = &analysis.Rule{Name: "osvscanner-exec-failed", Severity: analysis.Warning}
+	scanningParseFailure               = &analysis.Rule{Name: "osvscanner-parse-failed", Severity: analysis.Warning}
+	scanningSucceeded                  = &analysis.Rule{Name: "osvscanner-succeeded", Severity: analysis.Warning}
+	osvScannerCriticalSeverityDetected = &analysis.Rule{Name: "osvscanner-critical-severity-vulnerabilities-detected", Severity: analysis.Error}
+	osvScannerHighSeverityDetected     = &analysis.Rule{Name: "osvscanner-high-severity-vulnerabilities-detected", Severity: analysis.Warning}
+	osvScannerModerateSeverityDetected = &analysis.Rule{Name: "osvscanner-moderate-severity-vulnerabilities-detected", Severity: analysis.Warning}
+	osvScannerLowSeverityDetected      = &analysis.Rule{Name: "osvscanner-low-severity-vulnerabilities-detected", Severity: analysis.Warning}
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "osv-scanner",
+	Requires: []*analysis.Analyzer{sourcecode.Analyzer, archive.Analyzer},
+	Run:      run,
+	Rules: []*analysis.Rule{
+		missingOSVScanner,
+		osvScannerCriticalSeverityDetected,
+		osvScannerHighSeverityDetected,
+		osvScannerModerateSeverityDetected,
+		osvScannerLowSeverityDetected,
+		scanningFailure,
+		scanningParseFailure,
+		scanningSucceeded},
+}
+
+var scannerTypes = [...]string{
+	"go.mod",
+	"yarn.lock",
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	archiveFilesPath, ok := pass.ResultOf[archive.Analyzer].(string)
+	if !ok || archiveFilesPath == "" {
+		return nil, nil
+	}
+	sourceCodeDir, ok := pass.ResultOf[sourcecode.Analyzer].(string)
+	if !ok || sourceCodeDir == "" {
+		return nil, nil
+	}
+
+	// make sure osv-scanner is in PATH
+	ovsBinaryPath, err := exec.LookPath("osv-scanner")
+	if err != nil {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			missingOSVScanner,
+			"Binary for osv-scanner not found in PATH", "osv-scanner executable must be in your shell PATH.")
+		return nil, nil
+	} else {
+		missingOSVScanner.Severity = analysis.OK
+		if missingOSVScanner.ReportAll {
+			pass.ReportResult(
+				pass.AnalyzerName,
+				missingOSVScanner,
+				"Binary for osv-scanner was found in PATH", "osv-scanner executable exists in your shell PATH.")
+		}
+	}
+
+	// check for the different files we check for and run scanner on each
+	// if none are detected report OK
+	scanningPerformed := false
+	for _, scannerType := range scannerTypes {
+		lockFile := filepath.Join(sourceCodeDir, scannerType)
+		if _, err := os.Stat(lockFile); err == nil {
+			// perform scan
+			scanningPerformed = true
+			cmdArgs := []string{"--json", "--lockfile", lockFile}
+			data, err := exec.Command(ovsBinaryPath, cmdArgs...).Output()
+			// error output is expected from osv-scanner, but if the length is zero there was a problem
+			// running the command
+			if err != nil && len(string(err.Error())) == 0 {
+				pass.ReportResult(
```

Comment:
I prefer with return this error instead of reporting it on the validator results. That way we literally break the validator because of something that made the validator run.

Reply:
this now returns the error